src/Tmds.Ssh/SftpChannel.cs

```diff
@@ -1079,27 +1085,49 @@ private async ValueTask DownloadFileAsync(string remotePath, string localPath, l
             permissions = attributes.Permissions;
         }
 
-        using FileStream localFile = OpenFileStream(localPath, overwrite ? FileMode.Create : FileMode.CreateNew, FileAccess.Write, FileShare.None, permissions!.Value);
+        Stream? dstStream = null;
```

Comment:
We've lost the `using`.

You can do something like:
```cs
using FileStream? localFile = localPath is null ? null : OpenFileStream(...);
stream ??= localFile;
```

Reply:
Yes, I noticed the `finally` later as I read through the PR.
Can you change to adopt the pattern suggested above?

Reply:
yes will do